src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import I18n from 'i18n-js';
+import PropTypes from 'prop-types';
+import Events from '../../../utils/helpers/events';
+import DateHelper from '../../../utils/helpers/date';
+import DateValidator from '../../../utils/validations/date';
+import tagComponent from '../../../utils/helpers/tags';
+import DatePicker from './date-picker.component';
+import InputDecoratorBridge from '../input-decorator-bridge';
+import StyledDateInput from './date.style';
+import Textbox from '../textbox';
+
+const isoDateFormat = 'YYYY-MM-DD';
+const today = DateHelper.todayFormatted(isoDateFormat);
+
+/**
+ * A Date widget.
+ *
+ * == How to use a Date in a component:
+ *
+ * In your file
+ *
+ *   import Date from 'carbon-react/lib/components/Date';
+ *
+ * To render the Date:
+ *
+ *   <Date name="myDate" />
+ *
+ * @class Date
+ * @constructor
+ * @decorators {Input,InputIcon,InputLabel,InputValidation}
+ */
+class Date extends React.Component {
```

Comment:
Could this also be a candidate for a functional component?

Reply:
Yes it could, but I've already extracted the `DatePicker` part from that component and I've opted to leave it as it was, because of the amount of work that should be done to rewrite it to functional.

Reply:
I think the consensus is that we'll refactor components to functional in future tickets and focus on implementing the style changes for now